nixos/lib/make-options-doc/default.nix

```diff
@@ -153,16 +157,20 @@ in rec {
   # Convert options.json into an XML file.
   # The actual generation of the xml file is done in nix purely for the convenience
   # of not having to generate the xml some other way
-  optionsXML = pkgs.runCommand "options.xml" {} ''
+  optionsXML = pkgs.runCommand "options.xml" {
+    nativeBuildInputs = [ pkgs.nix.out ];
```

Comment:
```
diff --git a/nixos/lib/make-options-doc/default.nix b/nixos/lib/make-options-doc/default.nix
index efbd9a25508..4555da8ee04 100644
--- a/nixos/lib/make-options-doc/default.nix
+++ b/nixos/lib/make-options-doc/default.nix
@@ -158,7 +158,7 @@ in rec {
   # The actual generation of the xml file is done in nix purely for the convenience
   # of not having to generate the xml some other way
   optionsXML = pkgs.runCommand "options.xml" {
-    nativeBuildInputs = [ pkgs.nix.out ];
+    nativeBuildInputs = [ pkgs.nix ];
   } ''
     export NIX_STORE_DIR=$TMPDIR/store
     export NIX_STATE_DIR=$TMPDIR/state
diff --git a/nixos/modules/misc/documentation.nix b/nixos/modules/misc/documentation.nix
index 8c0689a75cc..26c598657d5 100644
--- a/nixos/modules/misc/documentation.nix
+++ b/nixos/modules/misc/documentation.nix
@@ -77,7 +77,7 @@ let
           pkgsLibPath = filter (pkgs.path + "/pkgs/pkgs-lib");
           nixosPath = filter (pkgs.path + "/nixos");
           modules = map (p: ''"${removePrefix "${modulesPath}/" (toString p)}"'') docModules.lazy;
-          nativeBuildInputs = [ pkgs.nix.out ];
+          nativeBuildInputs = [ pkgs.nix ];
         } ''
           export NIX_STORE_DIR=$TMPDIR/store
           export NIX_STATE_DIR=$TMPDIR/state
diff --git a/nixos/tests/installer.nix b/nixos/tests/installer.nix
index 398ad8de19c..b4e4e18b437 100644
--- a/nixos/tests/installer.nix
+++ b/nixos/tests/installer.nix
@@ -331,6 +331,7 @@ let
             libarchive.dev
             libxml2.bin
             libxslt.bin
+            nix.dev
             nixos-artwork.wallpapers.simple-dark-gray-bottom
             ntp
             perlPackages.ListCompare
```

less error prone if someone doesn't know why out is being referenced

Reply:
Would a short comment be better (e.g. `nativeBuildInputs uses the dev output, when we only need the binaries`), instead of pulling in the dev output needlessly?

Reply:
No, this is more future proof because what if someone adds nix to some other nativeBuildInputs